python/pyarrow/tests/test_compute.py

```diff
@@ -1567,8 +1568,10 @@ def test_round_to_multiple():
         assert pc.round_to_multiple(values, multiple,
                                     "half_towards_infinity") == result
 
-    with pytest.raises(pa.ArrowInvalid, match="multiple must be positive"):
-        pc.round_to_multiple(values, multiple=-2)
+    for multiple in [-2, pa.scalar(-10.4)]:
+        with pytest.raises(pa.ArrowInvalid,
+                           match="multiple must be nonnegative"):
+            pc.round_to_multiple(values, multiple=multiple)
```

Comment:
Can you also add a test for `TypeError` when the multiple is not convertible to a scalar?